### PR TITLE
feat: Do not show sign in note while loading user

### PR DESCRIPTION
**Pull request by mydea**
* Do not show signed in note until the user has loaded (or not).
* Extract the signed in check out (we'll need it for the org auth tokens as well)
* Make sure the sign in link does not open in a new tab, as the redirect back to docs already works 🎉

### CODE REVIEW

**Review by stephanie-anderson — APPROVED** [review]
awesome!!

**Review by shanamatthews — APPROVED** [review]
🙌 thank you for fixing this! It was driving me bonkers

File: src/components/codeContext.tsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import {createContext, useEffect, useState} from 'react';
 
+type CodeContextStatus = 'LOADING' | 'LOADED';
```
Comment (priscilawebdev) [inline src/components/codeContext.tsx:3]: nit: shouldn't this be lowercase?

File: src/components/signedInCheck.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React, {Fragment, useContext} from 'react';
+
+import {CodeContext} from './codeContext';
+
+export function SignedInCheck({
+  children,
+  ifIsAuthenticated,
+}: {
+  children: any;
```
Comment (priscilawebdev) [inline src/components/signedInCheck.tsx:9]:
```suggestion
  children: React.ReactNode;
```

File: src/components/signedInCheck.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  ifIsAuthenticated: boolean;
+}): JSX.Element {
+  const {codeKeywords, status} = useContext(CodeContext);
+
+  const user = codeKeywords.USER;
+
+  // Never render until loaded
+  if (status !== 'LOADED') {
+    return null;
+  }
```
Comment (priscilawebdev) [inline src/components/signedInCheck.tsx:19]:
```suggestion
  // Never render until loaded
  if (status !== 'LOADED') {
    return null;
  }
  
   const user = codeKeywords.USER;
```

File: src/components/signedInCheck.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React, {Fragment, useContext} from 'react';
+
+import {CodeContext} from './codeContext';
+
+export function SignedInCheck({
+  children,
+  ifIsAuthenticated,
```
Comment (priscilawebdev) [inline src/components/signedInCheck.tsx:7]:
```suggestion
  isUserAuthenticated,
```

🤷‍♀️

File: src/components/codeContext.tsx (hunk 1 above)
Comment (mydea) [reply src/components/codeContext.tsx:3]: good question, not sure about the conventions there 🤔 But I guess you're right, I'll make it lowercase! 👍

Comment (evanpurkhiser) [reply src/components/codeContext.tsx]: Why didn't we just make this `isLoading`?

File: src/components/codeContext.tsx [hunk 5]
```diff
@@ -80,6 +82,7 @@ type CodeContextType = {
   codeKeywords: CodeKeywords;
   sharedCodeSelection: any;
   sharedKeywordSelection: any;
+  status: CodeContextStatus;
```
Comment (evanpurkhiser) [inline src/components/codeContext.tsx:85]: Caan we have just done `isLoading`?

File: src/components/signedInCheck.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React, {Fragment, useContext} from 'react';
+
+import {CodeContext} from './codeContext';
+
+export function SignedInCheck({
```
Comment (evanpurkhiser) [inline src/components/signedInCheck.tsx:5]: This component feels kind of poorly named and under-documented.

Can we add a doc comment to this component to better describe what it does.

File: src/components/codeContext.tsx (hunk 1 above)
Comment (mydea) [reply src/components/codeContext.tsx]: I initially thought about wanting `error` state or something like this as well 🤔 ended up ignoring this for now. I can adjust this a bit.

File: src/components/signedInCheck.tsx (hunk 6 above)
Comment (mydea) [reply src/components/signedInCheck.tsx:5]: sure thing!

**Pull request merged**